core/arch/arm/plat-imx/a9_plat_init.S

```diff
@@ -56,6 +56,9 @@
 FUNC arm_cl2_config , :
 UNWIND(	.fnstart)
 
+	/* set_sp already setup sp */
+	push {lr}
```

Comment:
I'd push an even number of registers to keep the stack pointer well aligned. Please add an unwind directive too.